spec/models/annotation_category_spec.rb

```diff
@@ -34,6 +81,7 @@
       before do
         @row = []
         @row.push('annotation category name 2')
+        @row.push(nil)
```

Comment:
Raise error if upload of annotation category with same name as existing annotation category changes the criterion id. Apply to all scenarios. (three tests for possible id changes)

Reply:
Because the annotation text creation in case of either criterion being specified or the case where the criterion is not specified is a call to `build`, we know that the annotation texts will never be updated during an upload. They will only be created. (to address point 3)